### PR TITLE
Sprint 28 TLT-1849 Check send_messages_all permission

### PR DESCRIPTION
**Pull request by douglashall**
Depends on PRs:

https://github.com/penzance/canvas_python_sdk/pull/55
https://github.com/Harvard-University-iCommons/django-auth-lti/pull/29
https://github.com/Harvard-University-iCommons/django-icommons-common/pull/90

### CODE REVIEW

File: lti_emailer/requirements/base.txt [hunk 1]
```diff
@@ -9,8 +9,8 @@ hiredis==0.2.0
 django-cached-authentication-middleware==0.2.1
 django-angular==0.7.13
 huey==0.4.8
-git+ssh://git@github.com/penzance/canvas_python_sdk.git@v0.7.8#egg=canvas-python-sdk==0.7.8
 flanker==0.4.34
-git+ssh://git@github.com/Harvard-University-iCommons/django-icommons-common.git@v1.9.18#egg=django-icommons-common==1.9.18
+git+ssh://git@github.com/penzance/canvas_python_sdk.git@v0.7.8#egg=canvas-python-sdk==0.7.9
+git+ssh://git@github.com/Harvard-University-iCommons/django-icommons-common.git@v1.9.19#egg=django-icommons-common==1.9.18
```
Comment (rascalking) [inline lti_emailer/requirements/base.txt]: Need to update the version number at the end of the url to 1.9.19 as well.

File: lti_emailer/requirements/base.txt [hunk 2]
```diff
@@ -9,8 +9,8 @@ hiredis==0.2.0
...
 git+ssh://git@github.com/Harvard-University-iCommons/django-icommons-ui.git@v0.9.8#egg=django-icommons-ui==0.9.8
-git+ssh://git@github.com/Harvard-University-iCommons/django-auth-lti.git@v1.2.1#egg=django-auth-lti==1.2.1
+git+ssh://git@github.com/Harvard-University-iCommons/django-auth-lti.git@v1.2.2#egg=django-auth-lti==1.2.1
```
Comment (rascalking) [inline lti_emailer/requirements/base.txt]: Need to update the version number at the end of the url to 1.2.2 as well.

File: lti_emailer/requirements/base.txt [hunk 3]
```diff
@@ -9,8 +9,8 @@ hiredis==0.2.0
 django-cached-authentication-middleware==0.2.1
 django-angular==0.7.13
 huey==0.4.8
-git+ssh://git@github.com/penzance/canvas_python_sdk.git@v0.7.8#egg=canvas-python-sdk==0.7.8
 flanker==0.4.34
-git+ssh://git@github.com/Harvard-University-iCommons/django-icommons-common.git@v1.9.18#egg=django-icommons-common==1.9.18
+git+ssh://git@github.com/penzance/canvas_python_sdk.git@v0.7.8#egg=canvas-python-sdk==0.7.9
```
Comment (rascalking) [inline lti_emailer/requirements/base.txt]: Need to update the version number in the middle of the url to 0.7.9 as well.

File: mailgun/route_handlers.py [hunk 4]
```diff
@@ -60,20 +59,20 @@ def handle_mailing_list_email_route(request):
         return JsonResponse({'error': message}, status=406)  # Return status 406 so Mailgun does not retry
 
     # Always include teaching staff addresses with members addresses, so that they can email any list in the course
-    teaching_staff_addresses = ml.teaching_staff_addresses
-    member_addresses = teaching_staff_addresses.union(
-                           [m['address'] for m in ml.members])
+    try:
+        teaching_staff_addresses = ml.teaching_staff_addresses
+    except Exception as e:
+        logger.exception("Failed to find teaching staff addresses")
```
Comment (rascalking) [inline mailgun/route_handlers.py]: If you're intending to continue here, you need to do something like `teaching_staff_addresses = set()`, or the very next line will fail.

File: lti_emailer/requirements/base.txt (hunk 3 above)
Comment (douglashall) [inline lti_emailer/requirements/base.txt]: Thanks.

**Comment by rascalking** [comment]
:+1:

File: mailgun/route_handlers.py (hunk 4 above)
Comment (douglashall) [inline mailgun/route_handlers.py]: Oops, forgot to remove this debugging. Thanks.

**Pull request merged**